### DISCUSSION TITLE
installation on a LXC

### DISCUSSION THREAD

**Discussion by Rothammel** [blakeblackshear/frigate#1364] [Ask A Question]
can someone please explain to me how the installation works under a LX container?
https://blakeblackshear.github.io/frigate/installation#proxmox

**Comment by esmoyer** [comment]
So I was able to fumble through this and figure it out a few weeks ago. I have been trying to refine the instructions but now that it's working I don't want to mess with it to much! Hopefully this helps you get going. I compiled instructions from a few different sources to write myself some reminders/instructions on how I got it working. I have running Frigate in a Docker container in a Ubuntu LXC container on Proxmox. I have 4 cameras and 1 doorbell and so far it's been working well (at least as far as I can tell). 9ms inference time. Let me know if you have questions and I can try to help.

https://github.com/blakeblackshear/frigate/discussions/451
https://github.com/blakeblackshear/frigate/discussions/1111
https://medium.com/@konpat/usb-passthrough-to-an-lxc-proxmox-15482674f11d

1. Create new LXC Ubuntu and Install Docker

2. Install Coral Drivers on the proxmox node/host shell (not the Frigate LXC container):
	a. Add our Debian package repository to your system:

```
echo "deb https://packages.cloud.google.com/apt coral-edgetpu-stable main" | sudo tee /etc/apt/sources.list.d/coral-edgetpu.list

curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -

sudo apt-get update
```
	
3. Install the Edge TPU runtime:

```
sudo apt-get install libedgetpu1-std
	
sudo apt-get install python3-pycoral
	
mkdir coral && cd coral
git clone https://github.com/google-coral/pycoral.git
cd pycoral
	
bash examples/install_requirements.sh classify_image.py
	
python3 examples/classify_image.py \
--model test_data/mobilenet_v2_1.0_224_inat_bird_quant_edgetpu.tflite \
--labels test_data/inat_bird_labels.txt \
--input test_data/parrot.jpg
```
	
4. Via the proxmox host shell go to `/etc/pve/lxc` and edit the container file via `nano 10x.conf` (choose right number of your frigate LXC container.)
	
Put add the information below "swap:512" to your container config file making sure your USB bus numbers are set to where your USB is mounted on your system:

```
arch: amd64
cores: 4
features: nesting=1
hostname: frigatelxc
memory: 8192
nameserver: 192.168.1.90 192.168.1.91
net0: name=eth0,bridge=vmbr0,firewall=1,hwaddr=B2:72:05:93:9E:13,ip=dhcp,type=veth
onboot: 1
ostype: debian
rootfs: local-lvm:vm-106-disk-0,size=128G
searchdomain: domain.local
swap: 512
lxc.cgroup.devices.allow: c 226:0 rwm
lxc.cgroup.devices.allow: c 226:128 rwm
lxc.cgroup.devices.allow: c 29:0 rwm
lxc.cgroup.devices.allow: c 189:* rwm
lxc.apparmor.profile: unconfined
lxc.cgroup.devices.allow: a
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file 0, 0
lxc.mount.entry: /dev/bus/usb/002/ dev/bus/usb/002/ none bind,optional,create=dir 0, 0
lxc.cap.drop:
```

**Reply by Rothammel** [reply]
I thought the documentation here was talking about an installation without docker. currently, I have docker installed on the host, and frigate is running fine. but docker in an LXC creates overhead. 🤗

**Reply by esmoyer** [reply]
Oh gotcha...not sure about that! I guess I am not seeing any overhead with mine. My inferences are 9ms with the Coral and the CPU sits around 20ish% with just 2 cores. I suppose I would be interested in cutting out Docker if that were possible. Now I'm intrigued.
